### PR TITLE
[BUD-83] NavBar

### PR DESCRIPTION
**Pull request by michaelmoney**
# [BUD-83](https://digitalrig.atlassian.net/browse/BUD-83) Navbar

## Description

- Imported from MaterialUI
- Needs to be customised to match the screenshots
- Include hamburger icon
- Include back arrow

### CODE REVIEW

File: src/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import { AppBar, createStyles, Theme } from '@material-ui/core';
+import Toolbar from '@material-ui/core/Toolbar';
+import IconButton from '@material-ui/core/IconButton';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import { NavBarButton, NavBarProps } from './types';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
```
Comment (lucianojss) [inline src/components/NavBar/NavBar.tsx]: I cannot find where this class is being used, is this necessary?

File: src/components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      flexGrow: 1,
+    },
+    menuButton: {
+      marginRight: theme.spacing(2),
+    },
+    title: {
```
Comment (lucianojss) [inline src/components/NavBar/NavBar.tsx]: I cannot find where this class is being used, is this necessary?

File: src/components/NewbieSelect/__tests__/NewbieSelect.tsx [hunk 3]
```diff
@@ -1,8 +1,12 @@
 import React from 'react';
```
Comment (lucianojss) [inline src/components/NewbieSelect/__tests__/NewbieSelect.tsx]: Could you change the name of the file to `NewbieSelect.test.tsx` to keep consistency?

File: src/components/NavBar/NavBar.tsx (hunk 1 above)
Comment (michaelmoney) [reply src/components/NavBar/NavBar.tsx]: Good point: Fixed here 6968d4e7e8338a839706f0f60446463ecc90d6db and df0c5f9ab825a6913ccc7a4e0e1e2f2d2dc7f045

File: src/components/NavBar/NavBar.tsx (hunk 2 above)
Comment (michaelmoney) [reply src/components/NavBar/NavBar.tsx]: Good point: Fixed here 6968d4e7e8338a839706f0f60446463ecc90d6db and df0c5f9ab825a6913ccc7a4e0e1e2f2d2dc7f045

File: src/components/NewbieSelect/__tests__/NewbieSelect.tsx (hunk 3 above)
Comment (michaelmoney) [reply src/components/NewbieSelect/__tests__/NewbieSelect.tsx]: Done. This file wasn't even included in tests. 👍 
Fixed here 6968d4e7e8338a839706f0f60446463ecc90d6db and df0c5f9ab825a6913ccc7a4e0e1e2f2d2dc7f045

File: src/components/NavBar/__tests__/__snapshots__/NavBar.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,94 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Component - NavBar renders correctly with "back" type 1`] = `
+<header
+  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
+>
+  <div
+    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
+  >
+    <button
```
[the remark this answers is not in the document]
Comment (michaelmoney) [reply src/components/NavBar/__tests__/__snapshots__/NavBar.test.tsx.snap]: Good point: Fixed here 6968d4e7e8338a839706f0f60446463ecc90d6db and df0c5f9ab825a6913ccc7a4e0e1e2f2d2dc7f045

File: src/components/NavBar/__tests__/__snapshots__/NavBar.test.tsx.snap [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  </Toolbar>
+</AppBar>
+`;
+
+exports[`Component - NavBar renders correctly with "menu" type 1`] = `
+<AppBar
+  position="fixed"
+>
+  <Toolbar>
+    <IconButton
```
Comment (lucianojss) [inline src/components/NavBar/__tests__/__snapshots__/NavBar.test.tsx.snap]: It's not being mocked the `IconButton`

Comment (michaelmoney) [reply src/components/NavBar/__tests__/__snapshots__/NavBar.test.tsx.snap]: Hmmm... 🤔  
Line 7: ```jest.mock('@material-ui/core/IconButton', () => 'IconButton');```

**Pull request merged**